src/internal/puck/components/ThemeSidebar.tsx

```diff
@@ -1,7 +1,51 @@
+import { AutoFieldPrivate } from "@measured/puck";
 import React from "react";
+import { ThemeConfig } from "../../../utils/themeResolver.ts";
+import {
+  constructThemePuckFields,
+  constructThemePuckValues,
+} from "../../utils/constructThemePuckFields.ts";
 
-const ThemeSidebar = () => {
-  return <div></div>;
+type ThemeSidebarProps = {
+  savedThemeValues: any;
+  themeConfig?: ThemeConfig;
+  saveTheme: (theme: ThemeConfig) => void;
+};
+
+const ThemeSidebar = (props: ThemeSidebarProps) => {
+  const { saveTheme, themeConfig, savedThemeValues } = props;
+  if (!themeConfig) {
+    return <></>;
+  }
+
+  const handleChange = (themeCategoryKey: string, newValue: any) => {
+    const updatedConfig: ThemeConfig = {
+      ...themeConfig,
+      [themeCategoryKey]: {
+        ...themeConfig[themeCategoryKey],
+        styles: newValue,
+      },
+    };
+    saveTheme(updatedConfig);
+  };
+
+  return (
+    <div>
```

Comment:
Can you add this text at the top of the section? I'm thinking into one of those styled boxes. Alert seems like the best option in shadcn? https://ui.shadcn.com/docs/components/alert
> Changes to the theme will affect all templates and layouts for the entire site.